services/QuillLMS/spec/models/provider_classroom_delegator_spec.rb

```diff
@@ -70,4 +70,40 @@
       it { expect(subject.synced_status(student3.attributes)).to eq nil }
     end
   end
+
+  context 'canvas classroom' do
+    let(:canvas_instance) { create(:canvas_instance) }
+    let(:classroom) { create(:classroom, :from_canvas, canvas_instance: canvas_instance, students: students) }
+
+    let(:student1) { create(:user, :with_canvas_account, canvas_instance: canvas_instance) }
+    let(:student2) { create(:user, :with_canvas_account, canvas_instance: canvas_instance) }
+    let(:student3) { create(:user) }
+    let(:students) { [student1, student2, student3] }
+
+    before do
+      create(
+        :canvas_classroom_user,
+        :synced,
+        classroom_external_id: classroom.classroom_external_id,
+        user_external_id: student1.user_external_id(canvas_instance: canvas_instance)
+      )
+
+      create(
+        :canvas_classroom_user,
+        :unsynced,
+        classroom_external_id: classroom.classroom_external_id,
+        user_external_id: student2.user_external_id(canvas_instance: canvas_instance)
+      )
+    end
```

Comment:
Interesting to see this pattern as opposed to `let!`.  I don't know that I have a preference for either approach, but it's neat to see this.

Reply:
I tend to use before blocks over let! if I don't use the let! variable anywhere.